src/main/java/mods/railcraft/common/blocks/wayobjects/RoutingLogic.java

```diff
@@ -220,8 +238,8 @@ public AND(Condition a, Condition b) {
         }
 
         @Override
-        public boolean matches(IRoutingTile tile, EntityMinecart cart) {
-            return a.matches(tile, cart) && b.matches(tile, cart);
+        public int evaluate(IRoutingTile tile, EntityMinecart cart) {
+            return (a.evaluate(tile, cart) != 0 ? b.evaluate(tile, cart) : valueForBool(false));
```

Comment:
If a is false (0), it returns false; otherwise, it returns the value of b. That was my original plan, similar to how boolean short-circuit `and` and `or` work in some dynamically-typed languages, like Python and Lua.